src/components/molecules/ChatMessageBox/ChatMessageBox.tsx

```diff
@@ -25,13 +25,15 @@ export interface ChatMessageBoxProps {
   sendThreadReply: SendChatReply;
   unselectOption: () => void;
   isQuestion?: boolean;
+  closeThread: () => void;
 }
 
 export const ChatMessageBox: React.FC<ChatMessageBoxProps> = ({
   selectedThread,
   sendMessage,
   sendThreadReply,
   unselectOption,
+  closeThread,
```

Comment:
React's naming pattern for callbacks is with `on...` prefix. Modals have `onHide`, buttons have `onClick`, `onMouseEnter` etc... We should be following the same pattern when it comes to our components as well. If you rename this to `onReplyToThread` and I see the usage of the component I will not only know what it does but where it does it.

This shows what the prop does.
`<ChatMessageBox closeThread={closeThread}`

This shows where the prop is used for + what it does because of the function.
`<ChatMessageBox onReplyToThread={closeThread}`

@mike-lvov @0xdevalias Some input here? Do you agree with me.

Reply:
@denisdimitrov This is passing through an explicit action isn't it? This seems to align with the existing API here: `sendMessage, sendThreadReply, unselectOption`

Reply:
I haven't looked too deeply into this code, but I would probably have expected the `sendMessage` / `unselectOption` / etc to have been composed into a single handler 'higher up' in the app probably, rather than passing them all through just to compose them again within a handler here. But that's probably more of a refactor than we should be getting into there.

Reply:
> This is passing through an explicit action isn't it?

Yes, it is. But usually naming it with `on` prefix will also indicate where and how this action is actually used.

Reply:
If we create `ChatMessageButton` it should accept `onClick` not `sendMessage` prop. This should apply for all of our components not just buttons and modals.

Reply:
>I haven't looked too deeply into this code, but I would probably have expected the sendMessage / unselectOption / etc to have been composed into a single handler 'higher up' in the app probably

This is exactly what I mean, if the component itself accepts `onReply` then you will be 'forced' to create this handler higher and pass only a single function.

Reply:
@denisdimitrov refactored this part in 0a87b37